### PR TITLE
add gooey support

### LINKED ISSUE #57: add gooey support so it can be run with a gui
use regolith `helper_gui` as a template, though this implementation is more complicated than we will need here.

Not needed for the next release

### PR DESCRIPTION
**Pull request by yucongalicechen**
closes #57 
It's working for both CLI arguments and GUI. Here're some screenshots.
gui1
gui2
gui3

@sbillinge ready for some feedback!

### CODE REVIEW

File: src/diffpy/labpdfproc/labpdfprocapp.py [hunk 1]
```diff
@@ -1,140 +1,156 @@
...
+                "inputs are 'file.xy', 'data/file.xy', 'file.xy, data/file.xy', "
+                "'.' (load everything in the current directory), 'data' (load "
+                "everything in the folder ./data), 'data/file_list.txt' (load "
+                "the list of files contained in the text-file called "
+                "file_list.txt that can be found in the folder ./data), "
+                "'./*.chi', 'data/*.chi' (load all files with extension .chi in the "
+                "folder ./data)."
+            ),
+            "nargs": "+",
+            "widget": "MultiFileChooser",
```
[the remark this answers is not in the document]
Comment (sbillinge) [reply src/diffpy/labpdfproc/labpdfprocapp.py:36]: yes, if that is the case in teh gui let's have them as separate inputs

File: src/diffpy/labpdfproc/labpdfprocapp.py [hunk 2]
```diff
@@ -1,140 +1,156 @@
...
+    for arg in define_arguments():
+        kwargs = {key: value for key, value in arg.items() if key != "name"}
+        p.add_argument(*arg["name"], **kwargs)
+    args = p.parse_args()
+    return args
+
+
 def main():
-    args = get_args()
+    args = gooey_parser() if len(sys.argv) == 1 or "--gui" in sys.argv else get_args()
```
[the remark this answers is not in the document]
Comment (sbillinge) [reply src/diffpy/labpdfproc/labpdfprocapp.py:153]: Nicely done there.... the users will like it!

**Review by sbillinge — COMMENTED** [review]
this looks really great!  Congrats. If you make an issue to update the docs/tutorial I can merge this.  Or else we should put the updates to the docs on this PR and merge it together.

**Comment by sbillinge** [comment]
we have to find some kind of workaround to pin testing to py 3.12 until the Gooey dependency is updated.  This won't be so easy because of the centralized way we are doing testing at the moment, but let's try and find a way to just test this up to 3.12.

**Comment by bobleesj** [comment]
@sbillinge it appears that the latest conda-forge supports py310.. https://anaconda.org/conda-forge/gooey/files

Also, a similar issue that remains open:

https://github.com/chriskiehl/Gooey/issues/920

**Comment by sbillinge** [comment]
I am running gooey on Regolith in a 3.12 env.  So it is not a problem with the code per se, but the issue is how we run our tests and design our requirements.  ATM I think we are pulling a file from `release-docs` that is uniform across projects and we migrated at some point to it testing on 3.13, so downgrading labpdfproc will be a bit messy.....and we will have to think of how to re-update it later when gooey comes back up to speed (I volunteered on the project to do that but didn't hear back yet).

**Comment by yucongalicechen** [comment]
> I am running gooey on Regolith in a 3.12 env. So it is not a problem with the code per se, but the issue is how we run our tests and design our requirements. ATM I think we are pulling a file from `release-docs` that is uniform across projects and we migrated at some point to it testing on 3.13, so downgrading labpdfproc will be a bit messy.....and we will have to think of how to re-update it later when gooey comes back up to speed (I volunteered on the project to do that but didn't hear back yet).

I read online that we can use `matrix.python-version` to test different Python versions. I can try that. Basically it can test everything for one Python version, and everything else except for the incompatible package for another version.
I put everything we discussed here in issues #131, #132, #133.

**Comment by bobleesj** [comment]
> > I am running gooey on Regolith in a 3.12 env. So it is not a problem with the code per se, but the issue is how we run our tests and design our requirements. ATM I think we are pulling a file from `release-docs` that is uniform across projects and we migrated at some point to it testing on 3.13, so downgrading labpdfproc will be a bit messy.....and we will have to think of how to re-update it later when gooey comes back up to speed (I volunteered on the project to do that but didn't hear back yet).
> 
> I read online that we can use `matrix.python-version` to test different Python versions. I can try that. Basically it can test everything for one Python version, and everything else except for the incompatible package for another version. I put everything we discussed here in issues #131, #132, #133.

@yucongalicechen we have matrix testing configured https://github.com/diffpy/diffpy.labpdfproc/blob/main/.github/workflows/matrix-and-codecov-on-merge-to-main.yml in this repository. But, again, it's testing against 3.11, 3.12, 3.13. 

@sbillinge One possibility is that we could dynamically override default Python versions (3.11, 3.12, 3.13 in each repository:

From

```
jobs:
  matrix-coverage:
    uses: Billingegroup/release-scripts/.github/workflows/_matrix-and-codecov-on-merge-to-main.yml@v0
    with:
      project: diffpy.labpdfproc
      c_extension: false
      headless: false
```

to

```
jobs:
  matrix-coverage:
    uses: Billingegroup/release-scripts/.github/workflows/_matrix-and-codecov-on-merge-to-main.yml@v0
    with:
      project: diffpy.labpdfproc
      c_extension: false
      headless: false
      python_versions: 3.10, 3.11 # Override
```

**Comment by bobleesj** [comment]
@yucongalicechen 

```
Building wheels for collected packages: wxpython
  Building wheel for wxpython (pyproject.toml): started
  Building wheel for wxpython (pyproject.toml): finished with status 'error'
  error: subprocess-exited-with-error
```

seems like wxpython pip install fails here in the CI. Did you test pip install using conda env with Python 3.13 before?

**Comment by sbillinge** [comment]
> @yucongalicechen
> 
> ```
> Building wheels for collected packages: wxpython
>   Building wheel for wxpython (pyproject.toml): started
>   Building wheel for wxpython (pyproject.toml): finished with status 'error'
>   error: subprocess-exited-with-error
> ```
> 
> seems like wxpython pip install fails here in the CI. Did you test pip install using conda env with Python 3.13 before?

It gets complicated but we could conda install wx then pip install gooey, or just only test up to 3.12

**Comment by yucongalicechen** [comment]
> @yucongalicechen
> 
> ```
> Building wheels for collected packages: wxpython
>   Building wheel for wxpython (pyproject.toml): started
>   Building wheel for wxpython (pyproject.toml): finished with status 'error'
>   error: subprocess-exited-with-error
> ```
> 
> seems like wxpython pip install fails here in the CI. Did you test pip install using conda env with Python 3.13 before?

I just checked that Python 3.13 works on my computer

**Comment by bobleesj** [comment]
> I just checked that Python 3.13 works on my computer

Thanks. I will look more into this tmr!

**Comment by yucongalicechen** [comment]
> @yucongalicechen Let's try re-running it now. #136 is merged so that it the linux CI can build wxpython.

Working nicely!! Thanks Bob!!

**Comment by yucongalicechen** [comment]
@sbillinge I think this PR is ready to be merged. I've added issues #131 and #132 to take care of the docs and input argument.

**Comment by bobleesj** [comment]
@yucongalicechen it just takes a bit of time (28 mins) to build wxpython here for Linux.. (uses sdist instead of whl files since PyPI provides no linux whl file)

**Pull request merged**